cpp/include/cuvs/neighbors/detail/ivf_pq_helpers.hpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * raft::copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a raft::copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuvs/core/nvtx.hpp>
```

Comment:
No detail APIs in `include/`. These functions, while I think they seem strange in the public APIs, are used by the FAISS integration. We can keep these public for now, and maybe revisit in the future